File: java/backend-spring/src/main/java/com/countriesdb/examples/spring/ValidationController03.java
package com.countriesdb.examples.spring;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/validate")
public class ValidationController03 {
    private final CountriesDbClient client;

    public ValidationController03(CountriesDbClient client) {
        this.client = client;
    }

    // 3. Multiple country validation
    @PostMapping("/country/multiple")
    public ResponseEntity<?> validateMultipleCountries(@RequestBody Map<String, List<String>> payload) {
        List<String> countries = payload.get("countries");
        if (countries == null || countries.isEmpty()) {
            return ResponseEntity.badRequest().body(Map.of(
                "valid", false,
                "message", "Countries list is required"
            ));
        }
        
        Map<String, Object> result = client.validateCountry(Map.of("code", countries));
        
        @SuppressWarnings("unchecked")
        List<Map<String, Object>> results = (List<Map<String, Object>>) result.get("results");
        if (results != null) {
            for (Map<String, Object> item : results) {
                if (!Boolean.TRUE.equals(item.get("valid"))) {
                    return ResponseEntity.unprocessableEntity().body(result);
                }
            }
        }
        
        return ResponseEntity.ok(Map.of(
            "valid", true,
            "message", "All countries validated",
            "payload", payload
        ));
    }
}

File: csharp/backend-aspnetcore/Controllers/ValidationController03.cs
using Microsoft.AspNetCore.Mvc;
using System.Text.Json;

namespace BackendAspNetCore.Controllers;

[ApiController]
[Route("api/validate")]
public class ValidationController03 : ControllerBase
{
    private readonly CountriesDbClient _client;

    public ValidationController03(CountriesDbClient client)
    {
        _client = client;
    }

    // 3. Multiple country validation
    [HttpPost("country/multiple")]
    public async Task<IActionResult> ValidateMultipleCountries([FromBody] Dictionary<string, object> payload)
    {
        if (!payload.ContainsKey("countries") || payload["countries"] is not JsonElement countriesElement)
        {
            return BadRequest(new Dictionary<string, object>
            {
                ["valid"] = false,
                ["message"] = "Countries array is required"
            });
        }

        var countries = JsonSerializer.Deserialize<string[]>(countriesElement.GetRawText());
        if (countries == null || countries.Length == 0)
        {
            return BadRequest(new Dictionary<string, object>
            {
                ["valid"] = false,
                ["message"] = "At least one country code is required"
            });
        }

        var result = await _client.ValidateCountriesAsync(new Dictionary<string, object>
        {
            ["code"] = countries
        });

        if (result.Results == null)
        {
            return UnprocessableEntity(new ValidationResponse { Valid = false, Message = "Validation failed" });
        }

        foreach (var item in result.Results)
        {
            if (!item.Valid)
            {
                return UnprocessableEntity(new
                {
                    valid = false,
                    message = item.Message ?? "Invalid country code",
                    results = result.Results
                });
            }
        }

        return Ok(new Dictionary<string, object>
        {
            ["valid"] = true,
            ["message"] = "All countries validated",
            ["payload"] = payload
        });
    }
}

